pandas/core/strings.py

```diff
@@ -1020,7 +1023,17 @@ def str_extractall(arr, pat, flags=0):
 
     index = MultiIndex.from_tuples(index_list, names=arr.index.names + ["match"])
 
-    result = arr._constructor_expanddim(match_list, index=index, columns=columns)
+    # workaround #27953
```

Comment:
can you explain more? shouldn't the constructor handle this?

Reply:
Clarified the comment (We have to workaround unconditionally passing `dtype=arr.dtype`).